### PR TITLE
Added Ed25519 tests

### PR DESCRIPTION
**Pull request by melatron**
## Description of change

* Added Malleability test case.
* Added zip215 tests.
* Added ed25519 sign.input tests.
* Removed TODO that is not needed.
* Fixes issue #19 

## Type of change

- [x] Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

All existing and new tests pass.

## Change checklist

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by nothingismagick** [comment]
Copying @wollac's comment from previous PR
image

**Review by nothingismagick — CHANGES_REQUESTED** [review]
~~Please fix Clippy (even though it isn't your fault, its because of rust 1.50.0) - this is open source gardening :D~~

Clippy fixed here: https://github.com/iotaledger/crypto.rs/pull/52/files#diff-eb22024a611ef759fe3cbefcf751bf535d92d3154fc0e6641944cda7a82f654b

and also add a changefile:
https://github.com/iotaledger/crypto.rs/tree/dev/.changes

**Comment by nothingismagick** [comment]
For the reviewers: I have spoken with @melatron and he will setup gpg signing for his next PRs. I will sign off for this PR.

File: utils/test_vectors/py/ed25519_sign_transform.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    sig = binascii.unhexlify(x[3][0:128])
+
+    print("    TestVector {")
+    print(f"        secret_key: \"{bytes(sk).hex()}\",")
+    print(f"        public_key: \"{bytes(pk).hex()}\",")
+    print(f"        message: \"{msg.hex()}\",")
+    print(f"        signature: \"{sig.hex()}\",")
+    print("    },")
+
+print("]")
```
Comment (nothingismagick) [inline utils/test_vectors/py/ed25519_sign_transform.py]: Missing EOF linebreak

**Comment by melatron** [comment]
@Wollac About your questions from - https://github.com/iotaledger/crypto.rs/pull/53#issuecomment-780614782

The idea of leaving the tests there is that in Rust you write integration tests in the /tests folder and unit tests under the file for the specific functionality you are testing. For the tests that are in `src/ed25519.rs ` they also use `crate::test_utils` which is not listed as a feature in the Cargo.toml file. I guess the idea is to be used only internally. Because of that I figured that these tests are more like unit tests with internal use of `test_utils`.

**Comment by Wollac** [comment]
> Because of that I figured that these tests are more like unit tests with internal use of `test_utils`.

OK, fair enough. I was mainly wondering that they are kind of redundant or duplicates with the new reference test vectors. Let's leave them in then! more tests hardly ever hurt :slightly_smiling_face:

**Review by Wollac — APPROVED** [review]

**Review by nothingismagick — APPROVED** [review]

**Pull request merged**